src/core/execute.c

```diff
@@ -4262,10 +4262,10 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
         if (exec_context_restrict_namespaces_set(c)) {
                 _cleanup_free_ char *s = NULL;
 
-                r = namespace_flag_to_string_many(c->restrict_namespaces, &s);
+                r = namespace_flags_to_string_alloc(c->restrict_namespaces, &s);
                 if (r >= 0)
                         fprintf(f, "%sRestrictNamespaces: %s\n",
-                                prefix, s);
+                                prefix, strna(s));
```

Comment:
namespace_flags_to_string_alloc should alwasy set the output argument when returning >= 0. This is less error-prone. This strna should not be needed then.